### DISCUSSION TITLE
Make generic control of core actually generic

### DISCUSSION THREAD

**Discussion by dorthrithil** [vmware/clarity#5586] [Ideas] [answered]
Currently the only element that is working in `cds-control` is an input. Actually using a custom generic control element like in the example below leads to an error.

```html
<cds-control>
  <label>label text</label>
  <my-custom-control></my-custom-control>
  <cds-control-message>message text</cds-control-message>
</cds-control>
```

```
Uncaught (in promise) Error: input element is missing
```

**Comment by coryrylan** [comment, marked as answer]
This was recently fixed, and not yet published to the docs. You can add any generic content by adding the cds-control attirbute to the element replacing the input.

```html
<cds-control>
  <label>label</label>
  <p  cds-control cds-text="body" cds-layout="m-t:sm">
    Use the <code cds-text="code">cds-control</code> attribute to place generic content into a control layout.
  </p>
  <cds-control-message>control message</cds-control-message>
</cds-control>
```

- https://stackblitz.com/edit/typescript-i9brpq?file=index.html
- https://github.com/vmware/clarity/pull/5572
